File: C/1752-check-if-array-is-sorted-and-rotated/main.go
package main

import "fmt"

func check(nums []int) bool {
	idx := -1

	for i := 1; i < len(nums); i++ {
		if nums[i] < nums[i-1] {
			idx = i
			break
		}
	}
	if idx == -1 {
		return true
	}

	for i := 1; i < len(nums); i++ {
		ii := (idx + i) % len(nums)
		iii := (idx + i - 1) % len(nums)
		if nums[ii] < nums[iii] {
			return false
		}
	}

	return true
}

func main() {
	fmt.Println()
}

File: C/1752-check-if-array-is-sorted-and-rotated/main.rs
struct Solution {}

impl Solution {
  pub fn check(nums: Vec<i32>) -> bool {
    let mut idx: usize = 0;

    for i in 0..nums.len() - 1 {
      if nums[i] > nums[i + 1] {
        idx = i + 1;
      }
    }

    for i in 0..nums.len() - 1 {
      let mut idx1: usize = (i + idx) % nums.len();
      let mut idx2: usize = (idx1 + 1) % nums.len();
      if nums[idx1] > nums[idx2] {
        return false;
      }
    }
    true
  }
}
